xtask/src/codegen/gen_lint_completions.rs

```diff
@@ -12,39 +12,73 @@ pub(crate) fn generate_lint_completions() -> Result<()> {
         cmd!("git clone --depth=1 https://github.com/rust-lang/rust ./target/rust").run()?;
     }
 
-    let mut contents = String::from("use crate::completions::attribute::LintCompletion;\n\n");
-    generate_descriptor(&mut contents, "./target/rust/src/doc/unstable-book/src".into())?;
+    let mut contents = String::from(
+        r#"pub struct Lint {
+    pub label: &'static str,
+    pub description: &'static str,
+}
+
+"#,
+    );
+    generate_lint_descriptor(&mut contents)?;
+    contents.push('\n');
+
+    generate_feature_descriptor(&mut contents, "./target/rust/src/doc/unstable-book/src".into())?;
     contents.push('\n');
 
     cmd!("curl http://rust-lang.github.io/rust-clippy/master/lints.json --output ./target/clippy_lints.json").run()?;
     generate_descriptor_clippy(&mut contents, &Path::new("./target/clippy_lints.json"))?;
     let contents = reformat(&contents)?;
 
-    let destination =
-        project_root().join("crates/ide_completion/src/generated_lint_completions.rs");
+    let destination = project_root().join("crates/ide_db/src/helpers/generated_lints.rs");
     ensure_file_contents(destination.as_path(), &contents)?;
 
     Ok(())
 }
 
-fn generate_descriptor(buf: &mut String, src_dir: PathBuf) -> Result<()> {
-    buf.push_str(r#"pub(super) const FEATURES: &[LintCompletion] = &["#);
+fn generate_lint_descriptor(buf: &mut String) -> Result<()> {
+    let stdout = cmd!("rustc -W help").read()?;
+    let start = stdout.find("----  -------  -------").ok_or_else(|| anyhow::format_err!(""))?;
+    let end =
+        stdout.rfind("Lint groups provided by rustc:").ok_or_else(|| anyhow::format_err!(""))?;
+    buf.push_str(r#"pub const DEFAULT_LINTS: &[Lint] = &["#);
     buf.push('\n');
-    ["language-features", "library-features"]
+    let mut lints = stdout[start..end]
+        .lines()
+        .filter(|l| !l.is_empty())
+        .flat_map(|line| {
+            let (name, rest) = line.trim().split_once(char::is_whitespace)?;
+            let (_default_level, description) = rest.trim().split_once(char::is_whitespace)?;
+            Some((name.trim(), description.trim()))
+        })
+        .collect::<Vec<_>>();
+    lints.sort_by(|(ident, _), (ident2, _)| ident.cmp(ident2));
+    lints.into_iter().for_each(|(name, description)| push_lint_completion(buf, name, description));
+    buf.push_str("];\n");
+    Ok(())
+}
```

Comment:
Mmh, I thought this gave you all the lints rustc provides but it looks like its missing some like `unsafe_code` for example? Is there a better way to fetch the default lints?

Reply:
Oh wait, this seems to be my mistake somewhere

Reply:
`rustc -Whelp` shows `unsafe-code` for me.

Reply:
Ye I was grepping for `unsafe_code` not `unsafe-code` ...